app/Http/Controllers/API/LoginController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace App\Http\Controllers\API;
+
+use App\Http\Controllers\Controller;
+use App\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Validator as FacadesValidator;
+use Illuminate\Support\Facades\Log;
+
+class LoginController extends Controller
+{
+    public function login(Request $request) {
+        try {
+            $validator = $this->validator($request);
+
+            if ($validator->fails()) {
+                return response()->json([
+                    'success' => false,
+                    'data' => [
+                        'errors' => $validator->errors()
+                    ]
+                ]);
+            }
+
+            $data = $validator->validated();
+
+            if(Auth::attempt($data)) {
+                $user = User::where('email', $data['email'])->first();
+
+                $token = $user->createToken('user-'.$user->id)->accessToken;
+
+                return response()->json([
+                    'success' => true,
+                    'data' => [
+                        'user' => $user,
+                        'token' => $token
+                    ]
+                ]);
+            }
+
+            return response()->json([
+                'success' => false,
+                'data' => [
+                    'errors' => 'Invalid Credentials'
+                ]
+            ]);
+
+        } catch (\Exception $e) {
+            Log::error('An error occured while trying to authenticate the user: ' . $e->getMessage());
+            return response()->json([
+                'success' => false,
+                'data' => [
+                    'errors' => 'Login failed, please try again'
+                ]
+            ], 500);
+        }
+    }
+
+    /**
+     * Validate request
+     *
+     * @return \Illuminate\Contracts\Validation\Validator
+     */
+    private function validator(Request $request)
+    {
+        return FacadesValidator::make($request->all(), [
```

Comment:
@SaidiAli , it is  generally considered best-practise to select ONLY the request-parameters during authentication and user-registration scenerios in an application.
The approach to just "select-all" ($request->all()) request-parameters makes your code less verbose but more vulnerable to attacks. 
You may want to fix this.
```suggestion
        return FacadesValidator::make($request->only(['email', 'password']), [
```

Reply:
I will sir, in next PR.